pkg/cover/html.go

```diff
@@ -103,6 +105,184 @@ func (rg *ReportGenerator) DoHTML(w io.Writer, progs []Prog) error {
+			coveredInFunc += function.covered
+			pcsInFunc += function.pcs
+		}
+		totalLines := len(lines)
+		var coveredLines int
+		for _, line := range file.lines {
+			if len(line.progCount) != 0 {
+				coveredLines++
+			}
+		}
+		data = append(data, []string{
+			fname,
+			strconv.Itoa(coveredLines),
+			strconv.Itoa(totalLines),
+			strconv.Itoa(file.coveredPCs),
+			strconv.Itoa(file.totalPCs),
+			strconv.Itoa(totalFuncs),
+			strconv.Itoa(coveredInFunc),
+			strconv.Itoa(pcsInFunc),
+		})
+	}
+
+	return data, nil
+}
+
+func (rg *ReportGenerator) DoCSVFiles(w io.Writer, progs []Prog) error {
+	data, err := rg.convertToStats(progs)
+	if err != nil {
+		return err
+	}
+
+	sort.Slice(data, func(i, j int) bool {
+		if data[i][0] != data[j][0] {
+			return data[i][0] < data[j][0]
+		}
+		return data[i][1] < data[j][1]
+	})
+
+	writer := csv.NewWriter(w)
+	defer writer.Flush()
+	if err := writer.Write(csvFilesHeader); err != nil {
+		return err
+	}
+	return writer.WriteAll(data)
+}
+
+func groupCoverByFilePrefixes(datas [][]string, prefixes map[string][]string) map[string]map[string]string {
+	const zeroString = "0.00"
+	d := make(map[string]map[string]string)
+
+	for component, filters := range prefixes {
+		var coveredLines int
+		var totalLines int
+		var coveredPCsInFile int
+		var totalPCsInFile int
+		var totalFuncs int
+		var coveredPCsInFuncs int
+		var pcsInFuncs int
+		var percentLines string
+		var percentPCsInFile string
+		var percentPCsInFunc string
+
+		for _, prefix := range filters {
+			prefix = strings.TrimPrefix(prefix, "/")
+			for i := 0; i < len(datas); i++ {
+				data := datas[i]
+				if !strings.HasPrefix(data[0], prefix) {
+					continue
+				}
+				if s, err := strconv.Atoi(data[1]); err == nil {
+					coveredLines += s
+				}
+				if s, err := strconv.Atoi(data[2]); err == nil {
+					totalLines += s
+				}
+				if s, err := strconv.Atoi(data[3]); err == nil {
+					coveredPCsInFile += s
+				}
+				if s, err := strconv.Atoi(data[4]); err == nil {
+					totalPCsInFile += s
+				}
+				if s, err := strconv.Atoi(data[5]); err == nil {
+					totalFuncs += s
+				}
+				if s, err := strconv.Atoi(data[6]); err == nil {
+					coveredPCsInFuncs += s
+				}
+				if s, err := strconv.Atoi(data[7]); err == nil {
+					pcsInFuncs += s
+				}
+			}
+		}
+
+		if totalLines != 0 {
+			percentLines = fmt.Sprintf("%.2f", 100.0*float64(coveredLines)/float64(totalLines))
+		} else {
+			percentLines = zeroString
+		}
+		if totalPCsInFile != 0 {
+			percentPCsInFile = fmt.Sprintf("%.2f", 100.0*float64(coveredPCsInFile)/float64(totalPCsInFile))
+		} else {
+			percentPCsInFile = zeroString
+		}
+		if pcsInFuncs != 0 {
+			percentPCsInFunc = fmt.Sprintf("%.2f", 100.0*float64(coveredPCsInFuncs)/float64(pcsInFuncs))
+		} else {
+			percentPCsInFunc = zeroString
+		}
+
+		d[component] = map[string]string{
+			"component":  component,
+			"lines":      strconv.Itoa(coveredLines) + " / " + strconv.Itoa(totalLines) + " / " + percentLines + "%",
```

Comment:
Will be shorter and more readable to do as: `fmt.Sprintf("%v / %v / %.2f%%", coveredLines, totalLines, percent)`